lib/WeBWorK/PG/Translator.pm

```diff
@@ -1198,50 +1202,172 @@ sub process_answers{
+# 		$temp_ans  = $ans;
+# 		$temp_ans = '' unless defined($temp_ans); #make sure that answer is always defined
+# 		                                          # in case the answer evaluator forgets to check
+# 		####################################
+#  	    # get answer evaluator (old method)
+#  	    ####################################
+# 		
+# 		if ( defined($rh_correct_answers ->{$ans_name} ) ) {
+# 			$rf_fun  = $rh_correct_answers->{$ans_name};
+# 		} else {
+# 			warn "There is no answer evaluator for the question labeled $ans_name";
+# 		}
+# 		$self->{safe}->share('$rf_fun','$temp_ans');
+# 
+	    ####################################
+ 	    # gather answers and answer evaluator (new method)
+ 	    ####################################
+
+		local($new_rf_fun,$new_temp_ans) = (undef,undef);
+		my $answergrp = $PG->{PG_ANSWERS_HASH}->{$ans_name};  #this has all answer evaluators AND answer blanks (just to be sure ) 
+ 	    my $responsegrp = $answergrp->response_obj;
+#################
+# refactor to answer group?
+##################
+        $new_rf_fun = $answergrp->ans_eval;   
+        my ($ans, $errors) = $self->filter_answer( $responsegrp->get_response($ans_name) );     
+        $new_temp_ans = $ans; # avoid undefined errors in translator
+        my $skip_evaluation=0;
+        if (not defined($new_rf_fun) ) {
+        	$PG->warning_message( "No answer evaluator for the question labeled: $ans_name ");
+        	$skip_evaluation=1;
+        } elsif (not ref($new_rf_fun) =~ /AnswerEvaluator/ ) {
+				$PG->warning_message( "Error in Translator.pm::process_answers: Answer $ans_name: 
+				                    Unrecognized evaluator type |". ref($rf_fun). "|");
+				$skip_evaluation=1;
 		}
-		$temp_ans  = $ans;
-		$temp_ans = '' unless defined($temp_ans); #make sure that answer is always defined
-		                                          # in case the answer evaluator forgets to check
-		$self->{safe}->share('$rf_fun','$temp_ans');
- 	    
+		if (not defined($new_temp_ans) ) {
+				$PG->warning_message( "No answer blank provided for answer evaluator $ans_name ");
+				$skip_evaluation=1;
+		}
+		$PG->debug_message("Answers associated with $ans_name are $new_temp_ans ref=". ref($new_temp_ans) ) if defined $new_temp_ans and $local_debug;
+       #FIXME -- this is a hack for handling check boxes. 
+        if (ref( $new_temp_ans) =~/HASH/i) {
+        	my @tmp = ();
+        	foreach my $key (sort keys %$new_temp_ans) {
+        		push @tmp, $key if $new_temp_ans->{$key} eq 'CHECKED';
+        	}
+        	$new_temp_ans = join("\0",@tmp);
+        	$PG->debug_message("Hash answer associated with $ans_name is $new_temp_ans. ref=". ref($new_temp_ans)."<br/>".pretty_print($new_temp_ans) ) if $local_debug;
+        }
+        #FIXME -- hack to allow answers such as <4,6,7>  -- < and > have been escaped.
+        unless( $skip_evaluation) { 
+			$new_temp_ans =~ s/\&lt\;/</g; # <
```

Comment:
I don't think this is the right way to solve your encoding problem.  I think the answer is to move the `encode_pg_and_html` line here https://github.com/mgage/pg/blob/develop_translator_upd/macros/PGbasicmacros.pl#L364 to after the `RECORD_ANS_NAME` call.  I think this maybe should always have been the case but the PGanswergroup set up by `RECORD_ANS_NAME` wasn't used before. The answer value should be encoded before it is parsed by ev3 or printed to the screen, but after its been saved for processing. 

If this is the right thing to do then you will have to move encode_pg_and_html after `INSERT_RESPONSE` as well at
- https://github.com/mgage/pg/blob/develop_translator_upd/macros/PGbasicmacros.pl#L364
- https://github.com/mgage/pg/blob/develop_translator_upd/macros/PGbasicmacros.pl#L364